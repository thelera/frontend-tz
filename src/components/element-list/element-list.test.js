import configureStore from "redux-mock-store";
import ElementList from "./element-list.jsx";
import {Provider} from "react-redux";
import React from "react";
import renderer from "react-test-renderer";

const mockStore = configureStore([]);

const mock = [
  {
    name: `guitars`,
    children: [
      {
        name: `acoustic`,
        children: [
          {name: `Kremona`},
          {name: `Epiphone`},
          {name: `Gibson`},
          {name: `Yamaha`},
        ],
      },
      {
        name: `electric`,
        children: [
          {
            name: `Fender`,
            children: [
              {
                name: `Telecaster`,
              },
              {
                name: `Stratocaster`,
              },
              {
                name: `Jaguar`,
              },
            ],
          },
          {
            name: `Gibson`,
            children: [
              {
                name: `Les Paul`,
              },
              {
                name: `SG`,
              },
              {
                name: `ES-335`,
              },
              {
                name: `ES-339`,
              },
            ],
          },
        ],
      },
      {name: `acoustic bass`},
      {name: `electric bass`},
    ],
  },
];

it(`Render ElementList`, () => {
  const store = mockStore({});

  const tree = renderer
      .create(
          <Provider store={store}>
            <ElementList
              elements={mock}
            />
          </Provider>
      )
      .toJSON();

  expect(tree).toMatchSnapshot();
});
