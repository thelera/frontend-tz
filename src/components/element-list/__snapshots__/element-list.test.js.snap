// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ElementList 1`] = `
<ul
  className="element-list"
>
  <li
    className="element-list__item"
  >
    <div
      className="element-list__inner"
    >
      <h2
        className="element-list__title"
        onClick={[Function]}
      >
        guitars
      </h2>
      <button
        className="element-list__button element-list__button--show"
        onClick={[Function]}
      >
        Show path
      </button>
      <button
        className="element-list__button element-list__button--minus"
        onClick={[Function]}
      >
        Delete
      </button>
      <form
        className="element-list__form"
        onSubmit={[Function]}
      >
        <input
          className="element-list__input"
          placeholder="Type to add child node"
          type="text"
        />
        <button
          className="element-list__button element-list__button--plus"
          type="submit"
        >
          Add
        </button>
      </form>
    </div>
  </li>
</ul>
`;
